### PR TITLE
feat: index course_key and video_url in algolia

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
[PROD-3321](https://2u-internal.atlassian.net/browse/PROD-3321)
Adds course_key and video_url in algolia to be indexed for courses only.

### CODE REVIEW

File: course_discovery/apps/course_metadata/algolia_models.py [hunk 1]
```diff
@@ -419,6 +419,14 @@ def subscription_prices(self):
...
         return []
 
+    @property
+    def course_key(self):
+        return self.key
+
+    @property
+    def video_url(self):
+        return self.video.src if self.video else None
+
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/algolia_models.py]: How/What will be the values for Programs? Cause in Algolia, both Course & Program are part of same product index.

Comment (Ali-D-Akbar) [reply course_discovery/apps/course_metadata/algolia_models.py]: It's only defined for Course so Programs will have `null`

**Review by zawan-ila — APPROVED** [review]

File: course_discovery/apps/course_metadata/algolia_models.py (hunk 1 above)
Comment (DawoudSheraz) [reply course_discovery/apps/course_metadata/algolia_models.py]: - I would say to rename course_key to product_key and video_url to product_marketing_video_url
- Programs will have null --> Have you verified this by running indexing?

**Review by DawoudSheraz — COMMENTED** [review]
- Unit tests are missing.

File: course_discovery/apps/course_metadata/algolia_models.py (hunk 1 above)
Comment (Ali-D-Akbar) [reply course_discovery/apps/course_metadata/algolia_models.py]: - Okay
- yes https://www.algolia.com/apps/testing2MLYZ9WWU0/explorer/browse/product?query=demo&searchMode=search

**Review by DawoudSheraz — APPROVED** [review]
Once CI is green.

**Comment by mattdrayer** [comment]
🎆

**Pull request merged**